### ISSUE TITLE
[Host.AzureServiceBus] Exception thrown when provisioning topolgy of a shared subscription

### ISSUE THREAD

**Issue by EtherZa** [zarusz/SlimMessageBus#244]
When more than one consumer is configured to use the same subscription name with the Azure Service Bus transport, the `ServiceBusTopologyService` throws an exception when attempting to provision the subscription.

Sample configuration:

```cs
.Consume<AsbSample>(
    x => x.Topic("spike/slim")
        .SubscriptionName("one")
        .SubscriptionSqlFilter("CorrelationId = 'one'"))

.Consume<AsbSample2>(
    x => x.Topic("spike/slim")
        .SubscriptionName("one"))
```

Exception thrown in `ServiceBusTopologyService.ProvisionTopology()` as `x.SubscriptionName` is not unique.

![image](https://github.com/zarusz/SlimMessageBus/assets/1317506/c668a861-696d-400e-ad99-c149c5333482)

Messages are consumed, as expected, if the subscription already exists.

**Comment by zarusz** [comment]
@EtherZa are you willing to adjust that and make the PR?

**Comment by EtherZa** [comment]
> @EtherZa are you willing to adjust that and make the PR?

Yeah, sure. I'll see what I can whip up.

First thoughts:
1.  `ConsumerSettings` for a topic to be grouped by `SubscriptionName`.
2. All `CreateSubscriptionOptions` for the `SubscriptionName` need to be equal or throw `ConfigurationMessageBusException` if not.
3. Rules will be merged by name. Identical rules (name/filter/action) are fine; but rules with the same name but different filter criteria will throw `ConfigurationMessageBusException`.

**Comment by zarusz** [comment]
Yes, the settings need to be the same otherwise it might not make sense for users to register two consumers on the same subscription.

**Comment by zarusz** [comment]
Closed via #245
